tests/lib/rules/no-unneeded-ternary.js

```diff
@@ -46,14 +47,126 @@ ruleTester.run("no-unneeded-ternary", rule, {
             }]
         },
         {
+            code: "var a = x >= 2 ? true : false;",
+            output: "var a = x >= 2;",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 18
+            }]
+        },
+        {
+            code: "var a = x ? true : false;",
+            output: "var a = !!(x);",
```

Comment:
Disagree. This fix has too many assumptions built-in. We generally err on the side of doing fewer fixes rather than more, as we can never do damage by not changing code.